pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -625,49 +625,35 @@ func publishMetrics(ctx *zedagentContext, iteration int) {
 }
 
 func getDiskInfo(vrs types.VolumeRefStatus, appDiskDetails *metrics.AppDiskMetric) error {
+	actualSize, maxSize, err := utils.GetVolumeSize(vrs.ActiveFileLocation)
+	if err != nil {
+		return err
+	}
+	appDiskDetails.Disk = vrs.ActiveFileLocation
+	appDiskDetails.Used = RoundToMbytes(actualSize)
+	appDiskDetails.Provisioned = RoundToMbytes(maxSize)
 	if vrs.IsContainer() {
-		appDiskDetails.Disk = vrs.ActiveFileLocation
-		// XXX For container images, max size is coming zero
-		// from the controller. So for now, we are setting up
-		// total size equal to the used size.
-		appDiskDetails.Provisioned = RoundToMbytes(vrs.MaxVolSize)
-		appDiskDetails.Used = RoundToMbytes(vrs.MaxVolSize)
 		appDiskDetails.DiskType = "CONTAINER"
-	} else {
-		imgInfo, err := diskmetrics.GetImgInfo(vrs.ActiveFileLocation)
-		if err != nil {
-			return err
-		}
-		appDiskDetails.Disk = vrs.ActiveFileLocation
-		appDiskDetails.Provisioned = RoundToMbytes(imgInfo.VirtualSize)
-		appDiskDetails.Used = RoundToMbytes(imgInfo.ActualSize)
```

Comment:
We definitely want to return imgInfo.ActualSize. Does GetVolumeSize already do that?